File: apps/back-end/scripts/build-storage-image-index.ts
// import path from 'node:path';
// import { fileURLToPath } from 'node:url';

// import {
//   buildStorageImageIndex,
//   getDefaultStorageIndexPath,
//   writeStorageImageIndex,
// } from '../src/lib/storageImageIndex.js';

// const __filename = fileURLToPath(import.meta.url);
// const __dirname = path.dirname(__filename);

// const STORAGE_DIR = path.resolve(__dirname, '../storage');

// async function main() {
//   const index = await buildStorageImageIndex(STORAGE_DIR);
//   const indexPath = getDefaultStorageIndexPath(STORAGE_DIR);
//   await writeStorageImageIndex(STORAGE_DIR, index, indexPath);

//   const microCount = Object.keys(index.micro).length;
//   const normalCount = Object.keys(index.normal).length;
//   console.log(
//     `[done] index written: ${indexPath} (micro=${microCount}, normal=${normalCount}, generatedAt=${index.generatedAt})`
//   );
// }

// main().catch((error) => {
//   console.error('[fatal]', error);
//   process.exit(1);
// });
export {};

File: apps/py-tools/wiki-images-scrap/build_storage_image_index.py
from __future__ import annotations

import sys
from pathlib import Path

from storage_image_index import build_storage_image_index, get_default_storage_index_path, write_storage_image_index


SCRIPT_DIR = Path(__file__).resolve().parent


def main() -> None:
    """Build the local image index JSON from the current image scraper directory structure."""
    index = build_storage_image_index(SCRIPT_DIR)
    index_path = get_default_storage_index_path(SCRIPT_DIR)
    write_storage_image_index(SCRIPT_DIR, index, index_path)

    micro_count = len(index.micro)
    normal_count = len(index.normal)
    original_count = len(index.original)
    print(
        f"[done] index written: {index_path} "
        f"(micro={micro_count}, normal={normal_count}, original={original_count}, generatedAt={index.generatedAt})"
    )


if __name__ == "__main__":
    try:
        main()
    except Exception as error:
        print("[fatal]", error, file=sys.stderr)
        raise SystemExit(1) from error
